### DISCUSSION TITLE
HtmlPreviewVisitor handle text/calender separately

### DISCUSSION THREAD

**Discussion by TimBo93** [jstedfast/MimeKit#1118] [Ideas]
Hi  @jstedfast ,

thank you for your awesome library!

I use it this library to convert mails to html to display them in a browser.
Today we found e-mails with a `Content-Type: text/calendar; charset="utf-8"; method=REQUEST` section before a `text/html` section.

At this point:
https://github.com/jstedfast/MimeKit/blob/f2404c005f14711fb78c871fd7790081ad989d9f/UnitTests/HtmlPreviewVisitor.cs#L177 
we fill a body with an `.ics` file content instead of the actual html content.

My first patch would look like this:

```csharp
protected override void VisitTextPart(TextPart entity)
{
    TextConverter converter;

    if (body != null)
    {
        // since we've already found the body, treat this as an attachment
        attachments.Add(entity);
        return;
    }

    // we want to treat text/calendar as an attachment, not as a regular text part.
    if (entity.ContentType.IsMimeType("text", "calendar"))
    {
        calenderAttachments.Add(entity);
        return;
    }

    if (entity.IsHtml)
    {
...

```

where `calenderAttachments` is just a new list of `readonly List<MimeEntity> calenderAttachments = new List<MimeEntity>();`

I would love to know what your thoughts about this calender stuff are.

Kind regards
Tim

**Comment by jstedfast** [comment]
Yea, your modification is probably the best way to handle it.

**Reply by TimBo93** [reply]
nice! Thank you for your fast reply. 

Do you think it makes sense that I create a PR for the calendar case? Or is it a special edge case that doesn't necessarily need attention?

**Reply by jstedfast** [reply]
Sure, go ahead and submit a PR
